### ISSUE TITLE
Problems removing a local hook

### ISSUE THREAD

**Issue by teppokoivula** [processwire/processwire-issues#1067]
Hi Ryan.

I'm experiencing some weirdness while attaching a hook and then trying to remove it – long story short, it seems that this doesn't work for "local hooks" (i.e. those that are attached directly to an instance of a class):

```php
$this->addHook('sayHi', function($event) {
    echo "Hi!";
    $event->removeHook(null);
});
$this->sayHi();
$this->sayHi();
```

I would expect the hook method to run only on first request, but in fact it runs on both.

From the comments of WireHooks:

```
		// Note hookClass is always blank when this is a local hook
		$id = "$hookClass:$priority:$method";
```

And then in WireHooks::removeHook():

```
	public function removeHook(Wire $object, $hookID) {
		if(!empty($hookID) && strpos($hookID, ':')) {
			...
		}
		return $object;
	}
```

In other words it seems that removeHook() won't currently work for local hooks at all. If I change the code to `strpos($hookID, ':') !== false` the issue I'm observing gets resolved, but is there something else here that I'm missing?

**Comment by ryancramerdesign** [comment]
Thanks @teppokoivula that's a great find—fixed. Lots of people add hooks, but few ever remove them (since the duration is just the request), and local hooks aren't used very often either, so I'm guessing that unfortunately this one just hasn't come up before, and has likely been there a long time. This is really great that you found this and your example pointed out the issue perfectly. Glad to have this one fixed. Thanks!

**Comment by teppokoivula** [comment]
Thanks Ryan – looks good to me :)
